### ISSUE TITLE
p-tree lazy loading accessibility issue

### ISSUE THREAD

**Issue by IdoAmar** [primefaces/primeng#17960]
### Describe the bug

When navigating in lazy loaded p-tree with keyboard, opening the tree node by pressing right arrow opens the node correctly but focuses on the next node instead of the child node ( this probably happens because the nodes were not yet loaded and the code to focus on the next element does not wait for the results to be loaded before focusing on the next element)
Also i seen that in older versions of this component this behavior of focusing the first child when opening did not exist so this bug would not occur (the focus just stayed on the parent, which is probably better than focusing on a wrong element)

### Reason for not contributing a PR

- [x] Lack of time
- [ ] Unsure how to implement the fix/feature
- [ ] Difficulty understanding the codebase
- [ ] Other

### Reproducer

https://primeng.org/treeselect

### Environment

Chrome browser

### Angular version

^19.0.0

### PrimeNG version

v19

### Steps to reproduce the behavior

This occurs in every browser, to reproduce just go to the first example of lazy loaded tree and open with the right arrow the first element ("Node 0") the focus would go to sibling ("Node 1").

### Expected behavior

focus on first child ("Node 0-0") or staying in same element ("Node 0").

**Comment by IdoAmar** [comment]
This issue seem to stem from this code : 

![Image](https://github.com/user-attachments/assets/1e618d8a-8699-4e84-8982-3b05e280c88f)

the node is expanded but the data isnt loaded yet, the solution isnt trivial since the "lazy" boolean is for the scroll and not for data updates by expanding, so you cant just disable the settimeout if its lazy (for the simple solution of staying in the same element ("Node 0")
the other solution could be achieved with creating event when onchanges on data change but still, i may think there should be some type of input to indicate that the data is changing  so the current logic will stay the same and only when the data is a "changing data" this code solution will execute.
again currently theres no way to know it since "lazy" property is for scrolling not updating data in general
